### ISSUE TITLE
Api inconsistancies for spatial field response

### ISSUE THREAD

**Issue by robwent** [yiisoft/yii2#14532]
I'm not sure this is a bug, it may be more of an inconsistency.

If you have a mysql spatial field like `POINT(lat lng)` and you call the controller through `yii\rest\ActiveController;` as described in the docs, then the xml response will work and the spatial field will just be empty, but if you try json then you will get an error about a malformed response.

    {
	"name": "Exception",
	"message": "Malformed UTF-8 characters, possibly incorrectly encoded.",
	"code": 5,
	"type": "yii\\base\\InvalidParamException",
	"file": 
In my case, I just added the `public function fields()` to my main model, but I was previously trying to add this to the new rest controller as the docs didn't say where it should go (http://www.yiiframework.com/doc-2.0/guide-rest-resources.html#overriding-fields)
but what if you just want to get rid of a field for the api and not for everything else, do you need to create a new model under the api and override there?

**Comment by baizhiqiang** [comment]
I got this Exception "Malformed UTF-8 characters, possibly incorrectly encoded" when I  requested a base64 image stream api  in a rest controller action (extend yii\rest\Controller), now I have focused on the position of Exception that is $behaviors['contentNegotiator']['formats'] =[],I just clear the formater in my contorller and it works.
Anybody can tell me what is the best way to set the formater? sorry for my English.
